.github/workflows/wheels.yaml

```diff
@@ -102,21 +100,72 @@ jobs:
           nox --session libbezier-release
 
       - name: Build wheels
-        uses: pypa/cibuildwheel@932529cab190fafca8c735a551657247fa8f8eaf #v2.19.1
+        uses: pypa/cibuildwheel@932529cab190fafca8c735a551657247fa8f8eaf # v2.19.1
+        env:
+          BEZIER_INSTALL_PREFIX: ./.nox/.cache/libbezier-release/usr
+          CIBW_ARCHS: arm64
+          CIBW_BEFORE_BUILD: |
+            python -m pip install numpy
+          CIBW_BUILD: >-
+            cp310-*
+            cp311-*
+            cp312-*
+          CIBW_ENVIRONMENT: >-
+            MACOSX_DEPLOYMENT_TARGET=14.0
```

Comment:
This is really unfortunate but Homebrew installs binaries that target the current OS (which is 14.0)

This results in `delocate` failures when trying to target the minimum SDK version supported by Python. (M1 / arm64 first started in macOS 11.0 Big Sur)

```
...
  + delocate-wheel --require-archs arm64 -w /private/var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/cibw-run-lxw12ek6/cp310-macosx_arm64/repaired_wheel -v /private/var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/cibw-run-lxw12ek6/cp310-macosx_arm64/built_wheel/bezier-2023.7.29.dev1-cp310-cp310-macosx_11_0_arm64.whl
  INFO:delocate.delocating:Copying library /Users/runner/work/bezier/bezier/.nox/.cache/libbezier-release/usr/lib/libbezier.2023.7.28.dylib to bezier/.dylibs/libbezier.2023.7.28.dylib
  INFO:delocate.delocating:Copying library /opt/homebrew/Cellar/gcc/14.1.0_1/lib/gcc/current/libgfortran.5.dylib to bezier/.dylibs/libgfortran.5.dylib
  INFO:delocate.delocating:Copying library /opt/homebrew/Cellar/gcc/14.1.0_1/lib/gcc/current/libquadmath.0.dylib to bezier/.dylibs/libquadmath.0.dylib
  INFO:delocate.delocating:Copying library /opt/homebrew/Cellar/gcc/14.1.0_1/lib/gcc/current/libgcc_s.1.1.dylib to bezier/.dylibs/libgcc_s.1.1.dylib
  INFO:delocate.delocating:Modifying install name in bezier/.dylibs/libbezier.2023.7.28.dylib from /opt/homebrew/opt/gcc/lib/gcc/current/libgfortran.5.dylib to @loader_path/libgfortran.5.dylib
  INFO:delocate.delocating:Modifying install name in bezier/.dylibs/libgfortran.5.dylib from @rpath/libgcc_s.1.1.dylib to @loader_path/libgcc_s.1.1.dylib
  INFO:delocate.delocating:Modifying install name in bezier/_speedup.cpython-310-darwin.so from @rpath/libbezier.2023.dylib to @loader_path/.dylibs/libbezier.2023.7.28.dylib
  INFO:delocate.delocating:Modifying install name in bezier/.dylibs/libbezier.2023.7.28.dylib from /opt/homebrew/opt/gcc/lib/gcc/current/libquadmath.0.dylib to @loader_path/libquadmath.0.dylib
  INFO:delocate.delocating:Modifying install name in bezier/.dylibs/libgfortran.5.dylib from @rpath/libquadmath.0.dylib to @loader_path/libquadmath.0.dylib
  Fixing: /private/var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/cibw-run-lxw12ek6/cp310-macosx_arm64/built_wheel/bezier-2023.7.29.dev1-cp310-cp310-macosx_11_0_arm64.whl
  Traceback (most recent call last):
    File "/private/var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/cibw-run-lxw12ek6/cp310-macosx_arm64/build/venv/bin/delocate-wheel", line 8, in <module>
      sys.exit(main())
    File "/private/var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/cibw-run-lxw12ek6/cp310-macosx_arm64/build/venv/lib/python3.10/site-packages/delocate/cmd/delocate_wheel.py", line 110, in main
      copied = delocate_wheel(
    File "/private/var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/cibw-run-lxw12ek6/cp310-macosx_arm64/build/venv/lib/python3.10/site-packages/delocate/delocating.py", line 1004, in delocate_wheel
      out_wheel_fixed = _check_and_update_wheel_name(
    File "/private/var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/cibw-run-lxw12ek6/cp310-macosx_arm64/build/venv/lib/python3.10/site-packages/delocate/delocating.py", line 839, in _check_and_update_wheel_name
      raise DelocationError(
  delocate.libsana.DelocationError: Library dependencies do not satisfy target MacOS version 11.0:
  /private/var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/tmp0b6da8na/wheel/bezier/.dylibs/libbezier.2023.7.28.dylib has a minimum target of 13.0
  /private/var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/tmp0b6da8na/wheel/bezier/.dylibs/libgfortran.5.dylib has a minimum target of 14.0
  /private/var/folders/dm/88b38gj92jj53dgxdsm12qf00000gn/T/tmp0b6da8na/wheel/bezier/.dylibs/libquadmath.0.dylib has a minimum target of 14.0
```